### PR TITLE
[MeshingApplication][MMG] Disabling OMP in mmg getmetric call and adding a check in the tests

### LINKED ISSUE #7209: [MeshingApplication][MMG] Non-threadsafe operation in MMG process/utilities
Hi,

I have been trying to make use of the feature added in #5337, which returns the __true__ metric tensor after remeshing. 

This works fine if one thread is used:
*metric_tensor_2d (log scale) after remeshing with OMP_NUM_THREADS=1*
image

However, if I run with OMP_NUM_THREADS>1 the metric tensor is written wrongly.

*metric_tensor_2d (log scale) after remeshing with OMP_NUM_THREADS=4*
image

I tried the same  but removing the #pragma omp decorators both in:

https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MeshingApplication/custom_processes/mmg_process.cpp

https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

But the same behaviour persists. Any clue on what may be causing the issue?
__EDIT: I must have done something wrong when testing this. Testing it again after @roigcarlo comments, commenting the pragma in these two files prevents the issue from happening, so there is some OMP bug in the code.__

This can be reproduced with the following script:

(try running with OMP=1 and OMP=4 and take a look at the METRIC_SCALAR in the remeshed_gid_output )
[MMG_OMP_TestCase.zip](https://github.com/KratosMultiphysics/Kratos/files/4896901/MMG_OMP_TestCase.zip)

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
Closes #7209 

**Changelog**
- Removed Pragma OMP calls when calling `MMG2D_Get_scalarSol` `MMG2D_Get_vectorSol
` and similar.
- Adds a check in the test_remesh_rectangle from the MeshingApplication

**Additional info**
These library calls do not ask for a node Id to return a metric value, so it is important in the order that they are called. 
https://github.com/KratosMultiphysics/Kratos/blob/50620aca85f127b30ccdfa9e119dcd0758958890/applications/MeshingApplication/custom_utilities/mmg_utilities.cpp#L3061-L3068

The "SetMetricTensor" version asks for an Id so it is compatible with OMP. 

https://github.com/KratosMultiphysics/Kratos/blob/50620aca85f127b30ccdfa9e119dcd0758958890/applications/MeshingApplication/custom_utilities/mmg_utilities.cpp#L2865-L2876

### CODE REVIEW

**Comment by loumalouomega** [comment]
OK (It is more an mmg API problem then?)

**Review by loumalouomega — DISMISSED** [review]
OK

**Comment by marcnunezc** [comment]
@loumalouomega If I understand the usage correclty yes, but I will try to check this with them in the forum.

**Comment by marcnunezc** [comment]
Btw, @loumalouomega, right now we write the metric to our model_part twice __after__remeshing.

One in:

https://github.com/KratosMultiphysics/Kratos/blob/50620aca85f127b30ccdfa9e119dcd0758958890/applications/MeshingApplication/custom_utilities/mmg_utilities.cpp#L4087-L4124


And another in:

https://github.com/KratosMultiphysics/Kratos/blob/50620aca85f127b30ccdfa9e119dcd0758958890/applications/MeshingApplication/custom_processes/mmg_process.cpp#L276-L308


Do you know why we need the first one? Also, why is the check of the "OLD_ENTITY" needed in the second one?

File: applications/MeshingApplication/custom_processes/mmg_process.cpp [hunk 1]
```diff
@@ -274,7 +274,6 @@ void MmgProcess<TMMGLibrary>::ExecuteFinalize()
 
     // In case of considering meric tensor
     if (it_node_begin->Has(r_tensor_variable)) {
-        #pragma omp parallel for
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/mmg_process.cpp:277]: maybe add a comment that those loops cannot be done in parallel?

Comment (roigcarlo) [reply applications/MeshingApplication/custom_processes/mmg_process.cpp:277]: Or even better adding a test

**Comment by marcnunezc** [comment]
Hi, 

After talking to the MMG developers, they have confirmed that this is an issue on their side. They first wrote the Get() library calls without the need of specifying the Id and changing it right now would be challenging for them as it would break old code. They might do a separate library call or do a work around so that this call can be threadsafe, but for now we need to disable it. 

I have added comments on the for-loops as a warning and added the check on an exisiting MMG test (I confirmed it would fail for OMP>1 in the master)

Can someone reapprove?

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**